lib/core/text/flat.nit

```diff
@@ -1095,8 +1140,22 @@ redef class NativeString
 		var end_length = len
 		var pos = 0
 		var chr_ln = 0
-		while pos < len do
+		var rem = len
+		while rem > 0 do
 			var b = self[pos]
+			while rem > 4 do
+				var i = fetch_4_chars(pos)
+				if i & 0x80808080 != 0 then break
+				pos += 4
+				chr_ln += 4
+				rem -= 4
+			end
```

Comment:
on the last loop of this inner-while, if rem > 0 but rem < 4, then `b` has a wrong value, I'm not sure this is important but it seems fishy.